core/trino-main/src/main/java/io/trino/operator/DriverFactory.java

```diff
@@ -141,7 +145,7 @@ public synchronized void noMoreDrivers()
         }
     }
 
-    public synchronized boolean isNoMoreDrivers()
+    public boolean isNoMoreDrivers()
     {
         return noMoreDrivers;
```

Comment:
I'm pretty sure you need an annotation to say that it is ok to ignore the guarded check here (and some comments on why)

Reply:
You're right, the annotation is more correct over `operatorFactories` requiring the guarded access so I'll move the annotation there and leave the inline comment over `noMoreDrivers`.